internal/db/schema/options.go

```diff
@@ -0,0 +1,37 @@
+package schema
+
+// getOpts - iterate the inbound Options and return a struct.
+func getOpts(opt ...Option) options {
+	opts := getDefaultOptions()
+	for _, o := range opt {
+		o(&opts)
+	}
+	return opts
+}
+
+// Option - how Options are passed as arguments.
+type Option func(*options)
+
+// options = how options are represented
+type options struct {
+	withMigrationStates map[string]migrationState
+	withDeleteLog       bool
```

Comment:
It doesn't seem like this is ever used outside of tests...?

Reply:
that's correct... and without it doesn't appear you can write a test that only runs a subset of migrations.  Since my migration includes both DML and DDL, I need a way to run up migrations up to a specific migrations, then seed the database with test state, and then run my DML migration, and finally assert that the state is valid.